docs/content/develop/test.md

```diff
@@ -0,0 +1,154 @@
+   ```tf
+   resource "google_compute_subnetwork" "<%= ctx[:primary_resource_id] %>" {
+     name          = "<%= ctx[:vars]['subnetwork_name'] %>"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "<%= ctx[:vars]['network_name'] %>"
+     auto_create_subnetworks = false
+   }
+   ```
+5. Modify the relevant `RESOURCE_NAME.yaml` file under [magic-modules/mmv1/products](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/products) to include an [`examples`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/examples.rb) block with your test. For example:
+   ```yaml
+   examples:
+     - !ruby/object:Provider::Terraform::Examples
+       name: "compute_subnetwork_basic"
+       primary_resource_id: "example"
+       vars:
+         subnetwork_name: "example-subnet"
+         network_name: "example-network"
+   ```
+
+> **Warning:** Values in `vars` must include a `-` (or `_`). They [trigger the addition of a `tf-test` prefix](https://github.com/GoogleCloudPlatform/magic-modules/blob/6858338f013f5dc57729ec037883a3594441ea62/mmv1/provider/terraform/examples.rb#L244), which the sweeper uses to clean them up after tests run.
+6. If beta-only fields are being tested:
+   - Add `provider = google-beta` to every resource in the `*.tf.erb` config file.
+   - Add `min_version: beta` to the `examples` block in `RESOURCE_NAME.yaml`.
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+This section assumes you've used the [Add a resource]({{< ref "/develop/resource.md" >}}) guide to create your handwritten resource, and you have a working MMv1 config.
+
+> **Note:** If not, you can create one now, or skip this guide and construct the test by hand. Writing tests by hand can sometimes be a better option if there is a similar test you can copy from.
+
+1. Add the test in MMv1. Repeat for all the standard tests you will need.
+2. [Generate the providers]({{< ref "/get-started/generate-providers.md" >}}).
+3. From the provider, copy and paste the generated `*_generated_test.go` file into [`magic-modules/mmv1/third_party/terraform/tests`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) as a new file call `*_test.go`.
+4. Using an editor of your choice, remove the `Example` suffix from all function names.
+5. Modify the tests as needed.
+   - Remove the comments at the top of the file.
+   - If beta-only fields are being tested, do the following:
+     - Change the file suffix to `.go.erb`.
+     - Add `<% autogen_exception -%>` to the top of the file.
+     - Wrap the beta-only tests in a version guard: `<% unless version = 'ga' -%>...<% else -%>...<% end -%>`.
+{{< /tab >}}
+{{< /tabs >}}
+
+## Add an update test
```

Comment:
I might have to circle back to this comment after going through the other items. Could you clarify what you mean by scanning, I'm not sure I understand the use case?

Reply:
by "scan" I just mean "look at and be able to quickly identify / understand important pieces of information (for example because they are formatted consistently)"

Reply:
Hmm ok, I think that makes sense. It sounds like you are finding the `*` to be inconsistent with the other documentation, and would prefer a placeholder instead? A placeholder gets really weird in these types of instructions, and I'm not fully seeing how it improves on the scan-ability you mentioned, but I can try it out if you think that would fix the issue.

Reply:
coming back to this - I think I was thinking something like `resource_PRODUCT_RESOURCE_generated_test.go`. But I think I would also be fine with considering this non-blocking and coming back to it later if it's an issue.

Reply:
I've taken another pass at this, and think it has turned out better, but curious your thoughts.

Instead of having so many editing instructions in a row without providing a concrete starting or ending place, I added examples of the result. I think this helps give the reader a model of what is being changed (two main blocks of code), and breaks it up visually. It can probably still be tweaked, and we may even be able to trim down the instructions further because the reader has an example to compare to.

Regarding the placeholders, we can still fit them in here, but I did come across a piece that I found weird, which is capitalization. If we used placeholders, I would expect something like `resource_PRODUCT_RESOURCE_generated_test.go` and then further down `testAccCheckPRODUCT_RESOURCEDestroyProducer`, and besides the format looking weird, PRODUCT_RESOURCE is not the exact same value in both cases. Can you think of a good way to handle this?

Reply:
Ok, sounds good. FWIW I think that is a reasonable solution for the destroy producer, ideally if we can get the `ProductResource` part to clearly stand out, but I would be slightly worried about a reader thinking they need to delete a function with that exact name.

I will leave it with the asterisks for now as "good enough", but not opposed to future improvements.